### PR TITLE
Upgrade to webpack 2

### PR DESCRIPTION
**Pull request by winkerVSbecks**
- Upgraded to the webpack 2 toolchain
- Enabled tree-shaking
- Removed inject decorator
- Added instalbul threshold for CI 

The only weird thing is that I have to include `ts-helpers` in both `vendor.ts` and `app.ts` otherwise TS complaints. 

connects to rangle/rangle-starter#217

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,38 +45,44 @@
...
     "autoprefixer-core": "^6.0.1",
-    "awesome-typescript-loader": "^1.1.1",
-    "babel-core": "^6.9.1",
-    "babel-polyfill": "^6.9.1",
-    "babel-preset-es2015": "^6.9.0",
+    "awesome-typescript-loader": "^3.0.0-beta.9",
+    "babel-core": "^6.17.0",
+    "babel-loader": "^6.2.5",
+    "babel-polyfill": "^6.16.0",
+    "babel-preset-latest": "^6.16.0",
```
Comment (SethDavenport) [inline package.json]: Is there anyway we can do this without babel?

**Comment by SethDavenport** [comment]
Curious as to why we need Babel, but I realize that was inherited. Otherwise LGTM. :shipit:

File: package.json (hunk 1 above)
Comment (winkerVSbecks) [inline package.json]: Yes, I can look into an alternative TS compiler. I wasn't sure if `a-t` was required for coverage or not. Probably makes sense to do that in another PR.

Comment (SethDavenport) [inline package.json]: well normal usage of awesome-typescript-loader is to just use typescript. For some reason in our starters we have custom config to have ts output ES6 and then use babel to output ES5. But this is slow and complex.  I'm wondering if we can just go from TS -> ES5 directly, which is the default for that loader IIRC

**Pull request merged**